### PR TITLE
Fix flaky unit test

### PR DESCRIPTION
**Pull request by J-Thompson12**
Closes: #1199 

**What changed?**
Changed the interval time for pollAuthStatus. Sometimes it was checking the call count before making a call. 

**Why?**

**How did you test it?**

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Comment by J-Thompson12** [comment]
@jpellizzari good question. I couldnt replicate the issue after many runs when making this change. So I cannot tell you if it is fully fixed or less likely. I do know that I can make it fail 100% of the time by tweaking that number to 6. I believe it is fully fixed with this change.

**Comment by jpellizzari** [comment]
What about this issue @J-Thompson12 ?: https://github.com/benbjohnson/clock/issues/8

It has been fixed in a newer release than the one we currently have...

**Comment by dhwthompson** [comment]
> good question. I couldnt replicate the issue after many runs when making this change.

I think the underlying problem here, at least for me, is that it’s tricky to understand the sequence of events that the test is looking for, and where the race condition came from in the first place. I *think* the flaky test is coming from the fact that we’re setting the mock clock to the _exact_ time that we’d expect a retry, but it’s hard to be confident of that from the code itself. I can think of a few possible ways to address this:

1. More explanation in the test code, to make the relationships between values clearer. For instance, if we had `interval` set to 4 seconds, we could increment the current time by `interval + time.Second`, to make it clear that we want to be sure the interval has passed, and similarly .
2. A few more comments alongside the code, to make the intention clearer.
3. More context in the commit message. This has the benefit of not having to worry about space constraints (potentially even going full-on ASCII art timeline), but it’s further away from the code. I put a lot of stock in commit messages, but I don’t know yet how much they’re a part of the company culture here.

What do you think, @J-Thompson12? Do we have an opportunity here to make the intent of the code clearer, while also fixing a flaky test?

**Review by jpellizzari — APPROVED** [review]
:+1: Can we turn this test case back on:

https://github.com/weaveworks/weave-gitops/pull/1314/files#diff-70653d128987200191f5cf221d3925cc19806f8020ec8639dadb42883af1f221R94

**Comment by J-Thompson12** [comment]
@jpellizzari good catch. I didnt even see that was turned off.

File: pkg/services/auth/github_test.go [hunk 1]
```diff
@@ -104,16 +104,16 @@ var _ = Describe("Github Device Flow", func() {
...
 
-			// +5
-			c.Add(5 * time.Second)
+			// check one second after interval
+			c.Add(interval + 1*time.Second)
 			Expect(rt.calls).To(Equal(1), "should have tried the first time")
 
-			// +10
-			c.Add(5 * time.Second)
+			// check during back off
```
Comment (dhwthompson) [inline pkg/services/auth/github_test.go:111]: Nice. This is much easier for me to follow.

**Review by dhwthompson — APPROVED** [review]
Cool. Thanks for making those changes, @J-Thompson12.

When you merge this, would you be happy to squash all the commits down into a single commit, and put some of this context from our discussion into the commit message? If we have to come back to this later – if someone makes a change that re-flakifies the test, say, or if we want to do some refactoring – it’d be really helpful to have a bit more context as to what we reckon the problem was and how you solved it.

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/Masterminds/semver/v3 v3.1.1
-	github.com/benbjohnson/clock v1.1.0
```
[the remark this answers is not in the document]
Comment (J-Thompson12) [reply go.mod:7]: Yeah thats how I read that as well. Didnt hurt to update it though.

**Pull request merged**